src/main/kotlin/uk/gov/dluhc/emsintegrationapi/config/ApiProperties.kt

```diff
@@ -2,17 +2,32 @@ package uk.gov.dluhc.emsintegrationapi.config
 
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.stereotype.Component
+import javax.annotation.PostConstruct
 
 @Component
 class ApiProperties(
+    @Value("\${$REQUEST_HEADER_NAME}")
+    val requestHeaderName: String,
     @Value("\${$DEFAULT_PAGE_SIZE}")
     val defaultPageSize: Int,
-    @Value("\${$REQUEST_HEADER_NAME}")
-    val requestHeaderName: String
+    @Value("\${$MAX_PAGE_SIZE}")
+    val maxPageSize: Int,
 ) {
     companion object {
         private const val API_CONFIG_PREFIX = "dluhc"
         const val DEFAULT_PAGE_SIZE = "$API_CONFIG_PREFIX.default-page-size"
+        const val MAX_PAGE_SIZE = "$API_CONFIG_PREFIX.max-page-size"
         const val REQUEST_HEADER_NAME = "$API_CONFIG_PREFIX.request.header.name"
+        const val MAX_PAGE_SIZE_MIN_VALUE = 50
+        const val DEFAULT_PAGE_SIZE_MIN_VALUE = 10
+    }
+
+    @PostConstruct
+    fun validate() {
+        when {
+            maxPageSize < MAX_PAGE_SIZE_MIN_VALUE -> throw RuntimeException("The max page size must be greater than or equal to $MAX_PAGE_SIZE_MIN_VALUE")
```

Comment:
Just thinking if we should use a more specific Exception type here instead of using RuntimeException directly?

Reply:
As soon as runtime exception thrown the service won't start and we do not want handle it anywhere

Reply:
It is just a BAD REQUEST

Reply:
Just created a new ApplicationConfigurationException extended from RuntimeException

Reply:
@kristerbone : we configured the default page size max page size as environment variables, will more flexibility to service. The above validation is make sure that we configured a valid value to default page size and max page size.